### PR TITLE
Fix parsing of properties from command line in TestLauncher so that properties can contain "=" character

### PR DESCRIPTION
**Pull request by VladislavStefanov**

### CODE REVIEW

File: scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java [hunk 1]
```diff
@@ -101,18 +101,19 @@ public static LauncherProperties fromCommandLineArguments(String[] args) {
     for (String arg : args) {
       String[] propertyNameValuePair = arg.split(PROPERTY_NAME_VALUE_DELIMITER);
       ParamName propertyName = ParamName.fromPropertyName(propertyNameValuePair[0]);
-      String propertyValue = propertyNameValuePair[1];
 
       if(propertyName == null) {
         continue;
       }
 
+      String propertyValue = arg.substring(propertyName.value().length() + 1);
```
Comment (karaimin) [inline scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java:109]: You don't need this. The split method can accept second argument for the desired number of partitions starting from the beginning

**Review by karaimin — CHANGES_REQUESTED** [review]

File: scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java [hunk 2]
```diff
@@ -101,18 +101,22 @@ public static LauncherProperties fromCommandLineArguments(String[] args) {
...
     }
     return new LauncherProperties(properties);
   }
 
   private static boolean isArgsInValidFormat(String[] args) {
-    return Stream.of(args).allMatch(arg -> arg.split(PROPERTY_NAME_VALUE_DELIMITER).length == 2);
+    return Stream.of(args).allMatch(arg -> {
+      int delimiterPosition = arg.indexOf(PROPERTY_NAME_VALUE_DELIMITER);
+      return delimiterPosition >= 1 && delimiterPosition <= arg.length() - 2;
+    });
```
Comment (karaimin) [inline scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java:120]: same here

**Review by karaimin — CHANGES_REQUESTED** [review]

File: scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java [hunk 3]
```diff
@@ -99,12 +99,12 @@ public static LauncherProperties fromCommandLineArguments(String[] args) {
 
     Map<ParamName, LauncherProperty> properties = new EnumMap<>(ParamName.class);
     for (String arg : args) {
-      String[] propertyNameValuePair = arg.split(PROPERTY_NAME_VALUE_DELIMITER);
+      String[] propertyNameValuePair = arg.split(PROPERTY_NAME_VALUE_DELIMITER, 2);
```
Comment (karaimin) [inline scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java:102]: extract `2` to constant

File: scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java [hunk 4]
```diff
@@ -99,12 +99,12 @@ public static LauncherProperties fromCommandLineArguments(String[] args) {
...
       ParamName propertyName = ParamName.fromPropertyName(propertyNameValuePair[0]);
       String propertyValue = propertyNameValuePair[1];
 
       if(propertyName == null) {
-        continue;
+        throw new LauncherInitializationException("Unsupported property");
```
Comment (karaimin) [inline scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java:107]: I think that this check should be relocated to `isArgsInValidFormat`. Also, property name can be non-null but still unsupported

**Review by karaimin — CHANGES_REQUESTED** [review]

File: scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java (hunk 4 above)
Comment (VladislavStefanov) [reply scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java:107]: ParamName.fromPropertyName(propertyNameValuePair[0]) returns null if name is unsupported

Comment (karaimin) [reply scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java:107]: Ok, I agree

Comment (VladislavStefanov) [reply scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java:107]: If I relocate it I will be calling ParamName.fromPropertyName 2 times for every property. The format is valid, but the properties are unsupported. It returns null when unsupported.

File: scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java (hunk 3 above)
Comment (VladislavStefanov) [reply scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java]: Done

File: scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java (hunk 4 above)
Comment (karaimin) [reply scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/launcher/LauncherProperties.java:108]: It's not a big deal. You also, already split it twice.

**Review by karaimin — APPROVED** [review]

**Pull request merged**